addons/l10n_ch/migrations/11.2/pre-migrate.py

```diff
@@ -0,0 +1,13 @@
+# -*- coding: utf-8 -*-
+
+
+def migrate(cr, version):
+    cr.execute("SELECT res_id FROM ir_model_data WHERE module = 'l10n_ch' AND name='account_tax_report_line_chtax_solde_formula'")
+
+    expression_id = cr.fetchone()
+
+    if expression_id:
+        cr.execute(
+            "DELETE FROM account_report_external_value WHERE target_report_expression_id = %s", 
+            [expression_id]
```

Comment:
This doesn't work, `expression_id` is a tuple.
```suggestion
            [expression_id[0]]
```

Reply:
In module `ir_model_data` there is UNIQUE contraint (module, name). That is why I thought if there is `expression_id` then it must be single and I got it with `cr.fetchone()` , am I missing something here .

Reply:
As I wrote in the other comment, `cr.fetchone()` returns a row in the form of a tuple, with one element per column in the select clause. In your case, since you want the first and only column, you need to indicate that.

Reply:
I was not aware about the `fetchone()`'s returning is tuple.

Reply:
done

Reply:
> am I missing something here .

Yes, the output of fetchone is a row-tuple, even if it has only has one value. Now you are being lucky here because `(x)=x` is true when `x` is an int -- PG assumes the parenthesis means an integer-returning expression, instead of a tuple.

Example:
```
In [2]: self.env.cr.execute("SELECT id FROM res_users WHERE id=7")

In [3]: self.env.cr.fetchone()
Out[3]: (7,)

In [4]: self.env.cr.execute("SELECT id FROM res_users WHERE id=7")

In [5]: x=self.env.cr.fetchone()

In [6]: x
Out[6]: (7,)

In [7]: self.env.cr.mogrify("SELECT id FROM res_users WHERE id=%s", [x]).decode(
   ...: )
Out[7]: 'SELECT id FROM res_users WHERE id=(7)'
```

Reply:
Got it thank you for explanation